File: backend/Dtos/UserBlockDto.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Dtos
{

    public class UserBlockDto
    {
        public string BlockerId { get; set; } = string.Empty;
        public string BlockerName { get; set; } = string.Empty;
        public string BlockerUserName { get; set; } = string.Empty;
        public string? BlockerAvatarUrl { get; set; }
        public string BlockedId { get; set; } = string.Empty;
        public string BlockedName { get; set; } = string.Empty;
        public string BlockedUserName { get; set; } = string.Empty;
        public string? BlockedAvatarUrl { get; set; }
        public DateTime CreatedAt { get; set; }
    }
    
    public class UserBlockListDto
    {
        public string BlockedId { get; set; } = string.Empty;
        public string BlockedName { get; set; } = string.Empty;
        public string BlockedUserName { get; set; } = string.Empty;
        public string? BlockedAvatarUrl { get; set; }
        public DateTime CreatedAt { get; set; }
    }





}

File: frontend/src/app/dtos/userBlockDto.ts

export interface UserBlockDto {
  blockerId: string;
  blockerName: string;
  blockerUserName: string;
  blockerAvatarUrl: string | null;
  blockedId: string;
  blockedName: string;
  blockedUserName: string;
  blockedAvatarUrl: string | null;
  createdAt: string;
}

export interface UserBlockListDto {
  blockedId: string;
  blockedName: string;
  blockedUserName: string;
  blockedAvatarUrl: string | null;
  createdAt: string;
}